packages/cardpay-reward-programs/cardpay_reward_programs/rule.py

```diff
@@ -34,7 +34,7 @@ def set_user_defined_parameters(
         raise NotImplementedError
 
     @abstractmethod
-    def sql(self, table_query):
+    def sql(self, table_query, aux_table_query=None):
```

Comment:
this was extended to support requiring more than one table. There's a refactor coming that removes this requirement entirely using duckdbs "register" functionality which will also clean up the tests significantly but I want to do that alongside the shift to using the parquet metdata files.